clients/drcachesim/tracer/raw2trace.h

```diff
@@ -1273,6 +1281,18 @@ template <typename T> class trace_converter_t {
                 if (!error.empty())
                     return error;
             }
+            // TODO i#5934: This is a workaround for the trace invariant error triggered
+            // by consecutive duplicate system call instructions. Remove this when the
+            // error is fixed in the drmemtrace tracer.
+            if (instr->is_syscall() && impl()->get_last_pc_if_syscall(tls) == pc &&
+                instr_count == 1) {
+                impl()->add_to_statistic(tls, RAW2TRACE_STAT_DUPLICATE_SYSCALL, 1);
+                impl()->log(
+                    3, "Found block with duplicate system call instruction. Skipping.");
+                // Since this instr is in its own block, we're done.
```

Comment:
So the trace will have a timestamp+cpuid with nothing after it before the next timestamp?  I guess there's no invariant on that currently; maybe worth a comment that it's left in a slightly strange state.